### PR TITLE
Get version for current tree state not latest

### PR DESCRIPTION
**Pull request by licaon-kter**
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [-] I have updated the documentation accordingly.
- [-] I have added tests to cover my changes.
- [?] All new and existing tests passed.
- [x] This is **NOT** translation related. (See [here](https://github.com/aminecmi/ReaderforSelfoss/pull/170#issuecomment-355715654))

Eg. I want to build v17181133**2**1, I checkout v17181133**2**1, but the command yields v17181133**3**1.

Ref: https://gitlab.com/fdroid/fdroiddata/commit/54d39c04b9e1cb71e3318d7a2ee02bccf42b1451 and https://f-droid.org/wiki/page/apps.amine.bou.readerforselfoss/lastbuild_1718113321

### CODE REVIEW

**Comment by aminecmi** [comment]
Thanks for the PR.

I'll have to test this before merging.

I use tags to automate the build, the fdroid version generation and the play store deployment, and it saves so much time.

What happens if for example I have two tags on the same commit ?

**Comment by aminecmi** [comment]
Another question, why is fdroid trying to build an old version that isn't in the version.txt file ?

File: app/build.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@ buildscript {
 }
 
 def gitVersion() {
-    def process = "git for-each-ref refs/tags --sort=-authordate --format='%(refname:short)' --count=1".execute()
+    def process = "git describe --contains $(git rev-parse HEAD)".execute()
```
Comment (aminecmi) [inline app/build.gradle]: This does not work and causes build issues because of the `$`.

Comment (aminecmi) [reply app/build.gradle]: This `git describe --contains HEAD` does seem to do what you want.

Comment (licaon-kter) [reply app/build.gradle]: It worked for me locally, but feel free to modify it as needed, and I'll updated the metadata.

**Comment by licaon-kter** [comment]
Again, it builds for me [metadata](https://gitlab.com/fdroid/fdroiddata/blob/54d39c04b9e1cb71e3318d7a2ee02bccf42b1451/metadata/apps.amine.bou.readerforselfoss.txt) is updated per tag:
```
1718113321-github
1718113321
commit=v1718113321
```

So it needs to build this specific version.

Was my issue understood? 

> why is fdroid trying to build an old version that isn't in the version.txt file ?

It was not old, it was current *at the point in time* when fdroid update was run. If you published a new version in the mean time doesn't matter. WE shouldh be able to build that(any?) version that you've tagged in the past.

**Pull request closed without merging**